tornettools/generate_tgen.py

```diff
@@ -267,26 +268,68 @@ def __convert_privcount_key_to_tgen_key(str):
     else:
         return str
 
-def get_servers(args, n_clients):
+def __calculate_number_of_servers(args, n_public_clients, n_hs_clients):
+    # the number of servers will be a fraction of the number of clients
+    n_public_servers = n_public_clients * args.server_scale
+    n_hs_servers = n_hs_clients * args.server_scale
+    n_servers = n_public_servers + n_hs_servers
+
+    # if we don't have enough servers, scale both server counts
+    if n_servers < TGEN_SERVER_MIN_COUNT:
```

Comment:
@robgjansen @stevenengler I'm making some tweaks here to ensure neither server type can get rounded down to 0 (unless configured to use 100% one client type or the other).

More generally, should this minimum (10) be applied to each server type?

Reply:
Yeah, I think if we're gonna run HS servers, then it'd be good to have a handful so that the HS clients aren't _always_ going to the same server over and over. But note that I have no good reason for this limit other than "it seems a bit more resilient", so ymmv :)

Reply:
> More generally, should this minimum (10) be applied to each server type?

Sounds good to me, but should maybe add a comment that the behaviour of `TGEN_SERVER_MIN_COUNT` differs from `TGEN_CLIENT_MIN_COUNT`, since one will be a minimum for hs/non-hs servers separately, and the other is a minimum for the total.